packages/authorization-process/src/model/domain/apiConverter.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Client,
+  ClientKind,
+  Key,
+  KeyUse,
+  clientKind,
+  keyUse,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  ApiClient,
+  ApiClientWithKeys,
+  ApiClientKind,
+  ApiKey,
+  ApiKeyUse,
+} from "./models.js";
+import { missingUserId } from "./errors.js";
+
+export const ClientKindToApiClientKind = (kind: ClientKind): ApiClientKind =>
+  match<ClientKind, ApiClientKind>(kind)
+    .with(clientKind.consumer, () => "CONSUMER")
+    .with(clientKind.api, () => "API")
+    .exhaustive();
+
+export const KeyUseToApiKeyUse = (kid: KeyUse): ApiKeyUse =>
+  match<KeyUse, ApiKeyUse>(kid)
+    .with(keyUse.enc, () => "ENC")
+    .with(keyUse.sig, () => "SIG")
+    .exhaustive();
+
+export function clientToApiClient(
+  client: Client,
+  { includeKeys, showUsers }: { includeKeys: true; showUsers: boolean }
+): ApiClientWithKeys;
+export function clientToApiClient(
+  client: Client,
+  { includeKeys, showUsers }: { includeKeys: false; showUsers: boolean }
+): ApiClient;
+export function clientToApiClient(
+  client: Client,
+  { includeKeys, showUsers }: { includeKeys: boolean; showUsers: boolean }
+): ApiClientWithKeys | ApiClient {
+  return {
+    id: client.id,
+    name: client.name,
+    consumerId: client.consumerId,
+    users: showUsers ? client.users : [],
```

Comment:
I don't know if I like this, maybe the array should come already empty here

Reply:
@Carminepo2 I agree, it should belong to the logic, but we based this on the current [implementation](https://github.com/pagopa/interop-be-authorization-process/blob/1.0.x/src/main/scala/it/pagopa/interop/authorizationprocess/common/Adapters.scala#L30).

Reply:
+ 1, this may be a good improvement, same for `includeKeys`.
You can open an improvement task that can be implemented at a later time